xarray/core/variable.py

```diff
@@ -259,7 +259,7 @@ def _as_array_or_item(data):
 
     TODO: remove this (replace with np.asarray) once these issues are fixed
     """
-    data = data.get() if isinstance(data, cupy_array_type) else np.asarray(data)
+    data = np.asarray(data)
```

Comment:
FYI @jacobtomlinson this changes the behaviour with cupy arrays to make it consistent with other duck arrays: now to get at the data as numpy arrays you need to call `to_numpy()` instead of `.values`. (`.values` will instead call `np.asarray()` on the wrapped data, whatever it is.) This might break your cupy efforts - let me know if you don't think I should make this change!

Reply:
(I also don't have cupy on my system so have never actually running the tests added here that require cupy...)

Reply:
Thanks for the heads up.

Cupy tests pass locally for me with this change.

```
pytest xarray/tests/test_cupy.py
==================================================================== test session starts =====================================================================
platform linux -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1
rootdir: /home/jtomlinson/Projects/pydata/xarray, configfile: setup.cfg
plugins: forked-1.3.0, anyio-2.2.0, timeout-1.4.2, xdist-2.2.0, cov-2.11.1, asyncio-0.14.0
collected 3 items

xarray/tests/test_cupy.py ...                                                                                                                          [100%]

===================================================================== 3 passed in 2.40s ======================================================================
```